tensorboard/uploader/uploader.py

```diff
@@ -833,3 +837,41 @@ def _varint_cost(n):
         result += 1
         n >>= 7
     return result
+
+
+def _filter_graph_defs(events):
+    for e in events:
+        for v in e.summary.value:
+            if (
+                v.metadata.plugin_data.plugin_name
+                != graphs_metadata.PLUGIN_NAME
+            ):
+                continue
+            if v.tag == graphs_metadata.RUN_GRAPH_NAME:
+                data = list(v.tensor.string_val)
+                filtered_data = [_filtered_graph_bytes(x) for x in data]
+                filtered_data = [x for x in filtered_data if x is not None]
+                if filtered_data != data:
+                    new_tensor = tensor_util.make_tensor_proto(
+                        filtered_data, dtype=types_pb2.DT_STRING
+                    )
+                    v.tensor.CopyFrom(new_tensor)
+        yield e
+
+
+def _filtered_graph_bytes(graph_bytes):
+    try:
+        graph_def = graph_pb2.GraphDef().FromString(graph_bytes)
+    # The reason for the RuntimeWarning catch here is b/27494216, whereby
+    # some proto parsers incorrectly raise that instead of DecodeError
+    # on certain kinds of malformed input. Triggering this seems to require
+    # a combination of mysterious circumstances.
+    except (message.DecodeError, RuntimeWarning):
+        logger.warning(
+            "Could not parse GraphDef of size %d.", len(graph_bytes),
```

Comment:
Reinstate "Skipping" or similar in this warning, to clarify that the failure to parse means we don't upload.

Reply:
Meant to; thanks. Done.